compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingOps.cpp

```diff
@@ -175,6 +174,20 @@ getEncodingContractionDims(EncodingAttr encoding) {
   return linalg::inferContractionDims(indexingMaps);
 }
 
+std::string stringifyRoleIndex(IntegerAttr valueAttr) {
+  auto value = valueAttr.getValue().getZExtValue();
+  switch (value) {
+  default:
+    llvm_unreachable("invalid enum to stringify");
+  case MATMUL_LHS:
+    return "LHS";
+  case MATMUL_RHS:
+    return "RHS";
+  case MATMUL_RESULT:
+    return "RESULT";
```

Comment:
nit1: we usually put the `default` branch at the end of switch. Let's move it to the last branch for consistency.
nit2: we prefer using `assert(false)` in this case. It could potentially lead to UB. The only benefit we could have is small performance win, and it rarely happens. IREE project is following the style of using assertion. See https://github.com/iree-org/iree/pull/8521 for more details.

```suggestion
  case MATMUL_LHS:
    return "LHS";
  case MATMUL_RHS:
    return "RHS";
  case MATMUL_RESULT:
    return "RESULT";
  default:
    assert(false && "invalid index");
    return "";
```

Reply:
Good catch! not just potentially UB but explicitly UB. This example shows the kind of things that can then happen -- `return` instructions can be omitted at the end of the function (since the compiler is allowed to assume they would be unreachable) leading to execution silently continuing into the next function in the code segment.

https://godbolt.org/z/zdov8zj8a

Reply:
Didn't know that there was an discussion on this and indeed we shouldn't use it as assert here, and indeed this will result UB. Tried to wrap it inside a jump table and it is still [UB](https://godbolt.org/z/9rdfYqs13). 

So in this particular case it should be appended with `return "";` to avoid any UB, actually `assert` needs a following `return "";` here as well to avoid UB when assertions are disabled.

Also looked up in the code base and this pattern is a little bit abused here and there.

Updated the code.